src/charm.py

```diff
@@ -13,91 +13,189 @@
 """
 
 import logging
+from typing import Optional
 
-from ops.charm import CharmBase
+import yaml
+from ops.charm import CharmBase, PebbleReadyEvent, RelationEvent
 from ops.framework import StoredState
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import (
+    ActiveStatus,
+    BlockedStatus,
+    MaintenanceStatus,
+    RelationDataContent,
+    WaitingStatus,
+)
+from serialized_data_interface import (
+    NoCompatibleVersions,
+    NoVersionsListed,
+    get_interfaces,
+)
+
+from charms.nginx_ingress_integrator.v0.ingress import IngressRequires
+
+DEFAULT_BACKEND_STORE_URI = "sqlite:///mlflow.db"
+DEFAULT_ARTIFACT_ROOT = "./mlruns"
 
 logger = logging.getLogger(__name__)
 
 
 class MlflowCharm(CharmBase):
     """Charm the service."""
-
     _stored = StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
+        # set interfaces
+        try:
+            self.interfaces = get_interfaces(self)
+        except NoVersionsListed as error:
+            self.model.unit.status = WaitingStatus(str(error))
```

Comment:
To be honest, this is how it is used in several charms. But here is my explanation: 
Basically you are not able to change the version after deploying the charm and if versions are not same you should end up in block state, however, if you create a relation w/ a charm that is not yet ready, you end up in waiting state until the second charm is ready.

Reply:
It would help to add a comment before the `WaitingStatus` call to explain what it is being tried to get achieved there. I would understand the reason behind that is a transient error that will be resolved on next run.